### ISSUE TITLE
Problem with cups-filters - CUPS 2.4.2

### ISSUE THREAD

**Issue by pclinuxer** [OpenPrinting/cups-filters#585]
**Describe the bug**
CUPS 2.4.2 will not print, the cause being "Filter failed".

See here:
Problem: https://lists.cups.org/pipermail/cups/2024-May/075402.html
Solution: https://lists.cups.org/pipermail/cups/2024-May/075403.html

**To Reproduce**
Try to print (Samsung M2020W printer) in a Linux system with CUPS 2.4.2 installed.
If the `libcupsimage2` package is not installed, `cups-filters` will fail and CUPS will then remove the printer as shown in the last line of the `dmesg` output. 

```
~$ sudo dmesg
--- snip ---
usb 5-1: new high-speed USB device number 10 using ehci-pci
usb 5-1: New USB device found, idVendor=04e8, idProduct=3321, 
bcdDevice= 1.00
[usb 5-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
usb 5-1: Product: M2020 Series
usb 5-1: Manufacturer: Samsung Electronics Co., Ltd.
usb 5-1: SerialNumber: 0731B8GJBB029TD
usblp 5-1:1.0: usblp1: USB Bidirectional printer dev 10 if 0 alt 0 
proto 2 vid 0x04E8 pid 0x3321
usblp1: removed       # last line
~$
```
**Expected behavior**
The expected behaviour is the one I get once I install the `libcupsimage2` package.
ie: CUPS will print without any issues, printer is not removed.

**Screenshots**
N/A

**System Information:**
 - OS: Linux devuan 6.1.0-21-amd64
 - Browser: Firefox / palemoon / LibreWolf 
 - Version: various

**Additional context**
`CUPS 2.4.2-3+deb12u5` is the *.deb package present in both Debian Bookworm and Devuan Daedalus repositories.
It will not work properly if the `libcupsimage2` package is not installed.

```
~$ aptitude why libcupsimage2
i   cups			Suggests	foomatic-db-compressed-ppds | foomatic-db
p   foomatic-db			Recommends	printer-driver-all                       
p   printer-driver-all		Recommends	printer-driver-splix                     
p   printer-driver-splix	Depends		libcupsimage2 (>= 1.4.0)                 
~$ 
```
From what I can make of the above `aptitude` printout, cups suggests `foomatic-db` / `foomatic-db-compressed-ppds` which in turn recommends `printer-driver-all`.

In turn, `printer-driver-all` recommends `printer-driver-splix` but this last package **depends** on `libcupsimage2`.  

TL;DR
CUPS will not work properly because the `cups-filters` package fails if the `libcupsimage2` is not installed.  

It would seem (?) that `cups-filters` depends on `libcupsimage2` but it is not a dependency.
See here: https://packages.debian.org/bookworm/cups-filters

Does it make sense to have to install three unneeded packages (`foomatic-db-compressed-ppds` | `foomatic-db`, `printer-driver-all` and `printer-driver-splix`) to get `libcupsimage2` installed and CUPS 2.4.2 working properly?

But you know better than me with respect to all this, so please have a look.  8^)

Thanks in advance.

Best,

PCL
